federation/pkg/federation-controller/service/dns/dns_test.go

```diff
@@ -27,10 +27,19 @@ import (
 	"k8s.io/kubernetes/federation/apis/federation/v1beta1"
 	fakefedclientset "k8s.io/kubernetes/federation/client/clientset_generated/federation_clientset/fake"
 	"k8s.io/kubernetes/federation/pkg/dnsprovider/providers/google/clouddns" // Only for unit testing purposes.
+	"k8s.io/kubernetes/federation/pkg/federation-controller/service/ingress"
 	. "k8s.io/kubernetes/federation/pkg/federation-controller/util/test"
 	"k8s.io/kubernetes/pkg/api/v1"
 )
 
+// NewClusterWithRegionZone builds a new cluster object with given region and zone attributes.
+func NewClusterWithRegionZone(name string, readyStatus v1.ConditionStatus, region, zone string) *v1beta1.Cluster {
```

Comment:
yes. this is no longer required. I have another PR already complete which is based on top of this PR, which has removed this function. I am waiting for this PR to be merged to raise that PR.